src/CLI/CLI.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Main class to register WP-CLI commands.
+ *
+ * @package WooCommerce\Migrator\CLI
+ */
+
+namespace WooCommerce\Migrator\CLI;
+
+use WP_CLI;
+
+/**
+ * Main class for registering the WP-CLI commands.
+ */
+class CLI {
+
+	/**
+	 * The command registrar.
+	 *
+	 * @var Command_Registrar
+	 */
+	private $registrar;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Command_Registrar $registrar The command registrar.
+	 */
+	public function __construct( Command_Registrar $registrar ) {
+		$this->registrar = $registrar;
+	}
+
+	/**
+	 * Registers the WP-CLI commands.
+	 */
+	public function register_commands() {
+		if ( ! class_exists( 'WP_CLI' ) ) {
+			return;
+		}
+
+		$this->registrar->add_command(
+			'wc migrate products',
+			[ $this, 'products_command_handler' ],
```

Comment:
Re: separation of concerns:
Looking ahead, I would create a dedicated class or file for each command.